doc/how_to_guides/pilz_industrial_motion_planner/src/pilz_sequence.cpp

```diff
@@ -0,0 +1,283 @@
+  }
+  
+  moveit_visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to continue the demo");
+  
+  // [ --------------------------------------------------------------- ]
+  // [ ------------------ MoveGroupSequence Action ------------------- ]
+  // [ --------------------------------------------------------------- ]
+  // Plans and executes the trajectory
+
+  // MoveGroupSequence action client
+  using MoveGroupSequence = moveit_msgs::action::MoveGroupSequence;
+  auto client = rclcpp_action::create_client<MoveGroupSequence>(node, "/sequence_move_group");
+
+  // Verify that the action server is up and running
+  if (!client->wait_for_action_server(std::chrono::seconds(10))) {
+    RCLCPP_ERROR(LOGGER, "Error waiting for action server /sequence_move_group");
+    return -1;
+  }
+  
+  // Create a MotionSequenceRequest
+  moveit_msgs::msg::MotionSequenceRequest sequence_request;
+  sequence_request.items.push_back(item1);
+  sequence_request.items.push_back(item2);
+
+  // Create action goal
+  auto goal_msg = MoveGroupSequence::Goal();
+  goal_msg.request = sequence_request;
+
+  // Planning options
+  goal_msg.planning_options.planning_scene_diff.is_diff = true;
+  goal_msg.planning_options.planning_scene_diff.robot_state.is_diff = true;
+  // goal_msg.planning_options.plan_only = true;
+
+  // Goal response callback
+  auto send_goal_options = rclcpp_action::Client<MoveGroupSequence>::SendGoalOptions();
+  send_goal_options.goal_response_callback = [](std::shared_ptr<GoalHandleMoveGroupSequence> goal_handle) {
+    try {
+      if (!goal_handle) {
+        RCLCPP_ERROR(LOGGER, "Goal was rejected by server");
+      } else {
+        RCLCPP_INFO(LOGGER, "Goal accepted by server, waiting for result");
+      }
+    }
+    catch(const std::exception &e) {
+      RCLCPP_ERROR(LOGGER, "Exception while waiting for goal response: %s", e.what());
+    }
+  };
+
+  // Result callback
+  send_goal_options.result_callback = [](const GoalHandleMoveGroupSequence::WrappedResult &result) {
+    switch (result.code) {
+      case rclcpp_action::ResultCode::SUCCEEDED:
+        RCLCPP_INFO(LOGGER, "Goal succeeded");
+        break;
+      case rclcpp_action::ResultCode::ABORTED:
+        RCLCPP_ERROR(LOGGER, "Goal was aborted. Status: %d", result.result->response.error_code.val);
+        break;
+      case rclcpp_action::ResultCode::CANCELED:
+        RCLCPP_ERROR(LOGGER, "Goal was canceled");
+        break;
+      default:
+        RCLCPP_ERROR(LOGGER, "Unknown result code");
+        break;
+    }
+    RCLCPP_INFO(LOGGER, "Result received");
+  };
+
+  // Send the action goal
+  auto goal_handle_future = client->async_send_goal(goal_msg, send_goal_options);
+
+  // Get result
+  auto future_result = client->async_get_result(goal_handle_future.get());
+
+  // Wait for the result
+  if (future_result.valid()) {
+    auto result = future_result.get();  // Blocks the program execution until it gets a response
```

Comment:
same here, this needs to wait for the action to complete. You can use the same "loop and check if complete" paradigm from the service example, or just spin this node until the action is complete:

https://github.com/ros2/examples/blob/rolling/rclcpp/actions/minimal_action_client/not_composable.cpp#L44